Frameworks/SPMySQLFramework/TimeZone.m

```diff
@@ -0,0 +1,34 @@
+//
+//  TimeZone.m
+//  SPMySQL.framework
+//
+//  Created by Robin Kunde on 8/19/20.
+//
+
+#import "TimeZone.h"
+#import "SPMySQLStringAdditions.h"
+
+@implementation SPMySQLConnection (TimeZone)
+
+#pragma mark -
+#pragma mark Setting connection time zone
+
+- (BOOL)setTimeZone:(NSString *)timeZoneIdentifier
+{
+    // TODO: If the supplied time zone is already set, return success
+//    if ([encoding isEqualToString:theEncoding] && !encodingUsesLatin1Transport) {
+//        return YES;
+//    }
+
+    // Run a query to set the connection encoding
+    [self queryString:[NSString stringWithFormat:@"SET time_zone = %@", [timeZoneIdentifier mySQLTickQuotedString]]];
+
+    // If the query errored, no encoding change occurred - return failure.
+    if ([self queryErrored]) {
+        return NO;
+    }
+
+    return YES;
+}
```

Comment:
Need to handle the case where it's set to server timezone - what if someone changes from a timezone set to "null"?

Reply:
👍